### DISCUSSION TITLE
Matrix Issues

### DISCUSSION THREAD

**Discussion by skhull2** [MLB-LED-Scoreboard/mlb-led-scoreboard#508] [Q&A]
I have a 64w x 32h matrix but when I power on the board, everything is displayed 2x side by side.  I thought I followed the instructions for configuring the matrix, see attached:

![Screenshot from 2024-01-06 22-14-09](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/155863130/db8ca71b-0c60-4b2d-ba3b-70e6636d7efd)

[w64h32.json](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/files/13852211/w64h32.json)

Any help would be appreciated.

**Comment by ty-porter** [comment]
You need to specify the size of your matrix, it's size is otherwise inferred as 32x32 and the image can be multiplexed.

`python main.py --led-cols=64`

I'm not really sure what the screenshot is supposed to show.

**Reply by skhull2** [reply]
from the MLB-LED-Scoreboard directly, I ran the command sudo python main.py --led-cols=64

and got the message back:
=== snd_bcm2835: found that the Pi sound module is loaded. ===
Don't use the built-in sound of the Pi together with this lib; it is known to be
incompatible and cause trouble and hangs (you can still use external USB sound adapters).

See Troubleshooting section in README how to disable the sound module.
You can also run with --led-no-hardware-pulse to avoid the incompatibility,
but you will have more flicker.
Exiting; fix the above first or use --led-no-hardware-pulse

Still showing 32x32 matrix in multiplex.

in my previous screenshot I was showing that I had created a w64h32.json file in the coordinates folder, which is how i thought you configure for a 64w x 32h matrix.

any additional assistance with getting this configured for a 64x32 matrix would be appreciated.

**Reply by ty-porter** [reply]
The message you received is not relevant to the problem at hand. You can follow the steps that it provides if you want to increase the performance of the scoreboard once you have it configured correctly.

The 64x32 JSON configuration configuration is unlikely to be an issue here unless you are receiving specific error messages telling you otherwise.

If the scoreboard arguments do not resolve the issue you will need to dig further into the driver library we use. We do not maintain that code, so we can only provide suggestions. However, you can take a look at the flags we recommend trying here:

https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard?tab=readme-ov-file#flags

A good place to start would be `--led-gpio-mapping` depending on which adapter board (HAT, for instance) you may be using. Other common problems are hardware or wiring related.

A better resource is the rpi-rgb-led-matrix library that goes into much more detail about how this works than we do:

https://github.com/hzeller/rpi-rgb-led-matrix/blob/master/README.md

**Reply by skhull2** [reply]
thanks alot for the assistance.  I tired a few things as per the [rpi-rgb-led-matrix](https://github.com/hzeller/rpi-rgb-led-matrix) without luck.  I will continue to troubleshoot this.

**Comment by skhull2** [comment]
Still having issues with a 64x32 matrix.  I wonder if I need to wire both sides of the matrix?[
![17051940161798528556278716357190](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/155863130/6bd9119d-89d5-421e-8359-37b308b6f5e7)
](url)

**Comment by gotopes** [comment]
if you have raspberry pi 4b, the adafruit bonnet and 64x32 matrix, try this command line and also note whatever messages you get if any as there is a issue with new installs that has a fix that i was given today to try and it worked for me.

your matrix hookup looks right if you put the ribbon cable on the "data in" 16pin connector (which the arrow on the circuit board  might show).

first, let's try running from the mlb-led-scoreboard directory:

sudo python3 main.py --led-gpio-mapping="adafruit-hat" --led-rows=32 --led-cols=64 --led-brightness=05 --led-slowdown-gpio=4 --led-scan-mode=0

**Reply by skhull2** [reply]
![PXL_20240127_022640142](https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/assets/155863130/e19515ff-33de-4275-bb1c-b4375104152f)

The command works and the matrix is now showing 64x32, but when I close the Putty session, the matrix stops working

**Comment by gotopes** [comment]
sounds like progress is being made!  you might want to adjust the brightness thing to your liking between 0 and 100 (you probably find my setting of 5 too dim) and you might have to adjust the "--led-slowdown-gpio=4" to 0, 1, 2 or 3 if you get funky results at 4.  4 is where my 4b is best at, and it seems you are using a pi zero w, so your setting might be different.

i have not tried disconnecting putty ssh with the display running yet so i can't speak to the display stopping but i imagine if that happens to me too, the fix is that thing that makes the display autorun when the pi finishes booting up and i saw instructions for doing that somewhere on this project page, wiki, or newbie's guide.

**Comment by skhull2** [comment]
Here perhaps?
https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/wiki/Creating-a-systemd-startup-script

**Reply by gotopes** [reply]
yes!  have you tried this yet?  i haven't yet but i anticipate things going smoothly once i finish get my display working right with the command flags.  i have an occasional stray line across the display and it's not always in the same place so it's not dead led's, for instance.  i think i might try the bonnet pin 4/18 jumper and change the gpio mapping flag and see if that helps.

**Comment by skhull2** [comment]
Everything seems to be working great now.  Just need to find a nice
enclosure that will work for the matrix and my pi 4b

**Comment by skhull2** [comment]
I also completed the step to have it restart everyday at 4am
https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/wiki/Automatically-restarting-the-scoreboard-for-a-new-day

**Reply by skhull2** [reply]
Thanks a TON for your help!  My ultimate goal is to make this:
https://www.thingiverse.com/thing:3716753
